nrf_modem/doc/ug_nrf_modem_porting_os.rst

```diff
@@ -20,16 +18,149 @@ Arrows indicate that the elements can communicate with each other directly.
+The OS abstraction implementation is responsible for providing a dynamic memory allocation for both cases.
+
+Library Heap
+============
+
+The library heap is used by the modem for dynamic memory allocations.
+It has no location requirements and may use the OS provided heap implementation.
+
+Shared Memory
+=============
+
+The shared memory regions have a fixed size and need to reside in the lower 128kB of RAM.
+To be accessible by both the modem and the application, these regions must be configured as non-secure by the secure application.
+
+The following RAM overview diagram shows the placement of Modem library in the sequential RAM, and it also indicates the configurable memory position values.
+
+.. figure:: images/nrf_modem_memory.svg
+   :alt: Modem library memory overview
+
+   Modem library memory overview
+
+The minimum sizes of these regions are as follows:
+* The minimum size of the ctrl region is given in :file:`nrf_modem_platform.h`.
+* The RX/TX sizes are dependent on the use-case. They need to fit the data for the largest command or socket operation executed.
+  The Kconfig options :kconfig:option:`NRF_MODEM_LIB_SHMEM_TX_SIZE` and :kconfig:option:`NRF_MODEM_LIB_SHMEM_RX_SIZE` in the nRF Connect SDK can be taken as a reference.
+* The trace area size can be left zero if no traces are desired. Otherwise refer to :ref:`modem_trace`.
+
+Peripherals
+***********
+
+As the Modem library has been compiled to operate on peripherals in the non-secure domain, the following two peripherals must be configured to be non-secure:
+
+* NRF_IPC
+* NRF_POWER
+
+If you are using the hard-float variant of the Modem library, the FPU must be activated in both the secure domain and the non-secure domain, and must be configured to allow the non-secure application to run FPU instructions.
+
+The :file:`nrfx/mdk/system_nrf9160.c` file provides a template on how to configure the FPU in both cases.
+The system file also provides several Errata workarounds specific to the chip variant used, which are needed for any secure domain application.
+
+Faults and Traces
+*****************
+
+The Modem library provides facilities to obtain trace data and handle modem faults.
+Both features are optional and do not need to be used by the application.
+
+Handling of a modem fault
+=========================
+
+On initialization of the Modem library, the application registers a modem fault handler through the initialization parameters.
+If a fault occurs in the modem, the application is notified through the fault handler, which enables the application to read the fault reason (and in some cases the Modem's Program Counter) and take appropriate action.
+
+The following are the three alternatives to how the application can handle the modem fault:
+
+* The application logs the fault with no other action.
+* Restart the application.
+* Shutdown and reinitialize the modem and the Modem library.
```

Comment:
I think we should stick to two options:
- Re-initialize the modem via `nrf_modem_shutdown()` and `nrf_modem_init()` if the application can handle that 
- Reset the chip

Logging the fault and doing nothing is certainly an option but it's not practically useful (it's the default in NCS because of historical reasons).

Reply:
I would remove that part entirely and just link to the dedicated page.